src/db_adapters/accounts.rs

```diff
@@ -0,0 +1,157 @@
+use diesel::r2d2::{ConnectionManager, Pool};
+use diesel::{ExpressionMethods, PgConnection};
+use futures::join;
+use tokio_diesel::AsyncRunQueryDsl;
+use tracing::error;
+
+use near_indexer::near_primitives;
+
+use crate::models;
+use crate::schema;
+
+/// Saves new Accounts to database or deletes the ones should be deleted
+pub(crate) async fn handle_accounts(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    outcomes: &near_indexer::ExecutionOutcomesWithReceipts,
+) {
+    let successful_outcomes: Vec<near_indexer::IndexerExecutionOutcomeWithReceipt> = outcomes
+        .values()
+        .filter(|outcome_with_receipt| {
+            match outcome_with_receipt.execution_outcome.outcome.status {
+                near_primitives::views::ExecutionStatusView::SuccessValue(_)
+                | near_primitives::views::ExecutionStatusView::SuccessReceiptId(_) => true,
+                _ => false,
+            }
+        })
+        .cloned()
+        .collect();
+
+    let store_accounts_future = store_accounts(&pool, &successful_outcomes);
+    let remove_accounts_future = remove_accounts(&pool, &successful_outcomes);
+
+    join!(store_accounts_future, remove_accounts_future);
+}
+
+async fn store_accounts(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    outcomes: &[near_indexer::IndexerExecutionOutcomeWithReceipt],
+) {
+    let accounts_to_create: Vec<models::accounts::Account> = outcomes
+        .iter()
+        .filter_map(|outcome_with_receipt| {
+            if let Some(receipt) = &outcome_with_receipt.receipt {
+                match &receipt.receipt {
+                    near_primitives::views::ReceiptEnumView::Action { actions, .. } => {
+                        let accounts: Vec<models::accounts::Account> = actions
+                            .iter()
+                            .filter_map(|action| {
+                                if let near_primitives::views::ActionView::CreateAccount = action {
+                                    Some(models::accounts::Account::new(
+                                        receipt.receiver_id.to_string(),
+                                        &receipt.receipt_id,
+                                    ))
+                                } else {
+                                    None
+                                }
+                            })
+                            .collect();
+                        Some(accounts)
+                    }
+                    _ => None,
+                }
+            } else {
+                None
+            }
+        })
+        .flatten()
```

Comment:
A single receipt can only create a single account (having multiple `CreateAccount` actions in a single receipt is pointless):

```suggestion
        .filter_map(|outcome_with_receipt| outcome_with_receipt.receipt)
        .filter_map(|receipt| if let near_primitives::views::ReceiptEnumView::Action { actions, .. } = receipt.receipt {
            Some((receipt, actions))
        } else {
            None
        })
        .filter_map(|(receipt, actions)| actions
            .iter()
            .filter(|action| matches!(action, near_primitives::views::ActionView::CreateAccount))
            .map(|_create_account_action| models::accounts::Account::new(
                receipt.receiver_id.to_string(),
                &receipt.receipt_id,
            )
            .next()
        )
```

Reply:
Just to clarify. Pointless or impossible?

Reply:
I don't know about impossibility, but it is clearly pointless.